File: kotlin/llm/src/commonMain/kotlin/dev/deviceai/llm/LlmGenConfig.kt
package dev.deviceai.llm

/**
 * Per-request generation parameters.
 * System prompt is passed as [LlmMessage] with role [LlmRole.SYSTEM] instead.
 *
 * @param maxTokens Maximum tokens to generate per request (default 512)
 * @param temperature Sampling temperature — higher = more creative (default 0.7)
 * @param topP Nucleus sampling probability threshold (default 0.9)
 * @param topK Top-K sampling limit (default 40)
 * @param repeatPenalty Penalty for repeating tokens (default 1.1)
 */
data class LlmGenConfig(
    val maxTokens: Int = 512,
    val temperature: Float = 0.7f,
    val topP: Float = 0.9f,
    val topK: Int = 40,
    val repeatPenalty: Float = 1.1f
)

File: swift/Sources/DeviceAiLlm/Config/LlmGenConfig.swift
/// Per-request generation parameters. Mirrors `LlmGenConfig` in kotlin/llm.
public struct LlmGenConfig: Sendable {
    public var maxTokens:     Int    = 512
    public var temperature:   Float  = 0.7
    public var topP:          Float  = 0.9
    public var topK:          Int    = 40
    public var repeatPenalty: Float  = 1.1
    public var ragStore:      (any Retrieving)? = nil
    public var ragTopK:       Int    = 3

    public init(maxTokens: Int = 512, temperature: Float = 0.7, topP: Float = 0.9,
                topK: Int = 40, repeatPenalty: Float = 1.1,
                ragStore: (any Retrieving)? = nil, ragTopK: Int = 3) {
        self.maxTokens = maxTokens; self.temperature = temperature
        self.topP = topP; self.topK = topK; self.repeatPenalty = repeatPenalty
        self.ragStore = ragStore; self.ragTopK = ragTopK
    }
}
